test/unittests/t_cache_ram.cc

```diff
@@ -0,0 +1,347 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <stdint.h>
+#include <alloca.h>
+#include <gtest/gtest.h>
+
+#include "hash.h"
+#include "cache.h"
+#include "cache_ram.h"
+#include "statistics.h"
+
+using namespace std;  // NOLINT
+using cache::RamCacheManager;
+
+static const unsigned cache_size = 1024;
+static const unsigned alloc_size = 16;
+
+
```

Comment:
I think it makes sense to add a text fixture to do the common test case initialization, e.g. the `RamCacheManager`.  The unit test needs to live in the same name space as the production code in order to make the gtest macro magic work.